src/components/products-list/product-card.tsx

```diff
@@ -10,10 +19,85 @@ export interface ProductCardProps {
   price: string;
   state: string;
   image_url: string;
+  isFavorite: boolean;
+  refetch: () => void;
 }
 
+export type TitleAndFavoriteProps = {
+  id: number;
+  name: string;
+  isFavorite: boolean;
+  refetch: () => void;
+};
+
+const TitleAndFavorite = ({
+  id,
+  name,
+  isFavorite,
+  refetch,
+}: TitleAndFavoriteProps) => {
+  const { mutate: addFavorite } = useAddFavorite({
+    onSuccess: () => {
+      refetch();
```

Comment:
I recommend you to read about queries invalidations: https://tanstack.com/query/v3/docs/framework/react/guides/invalidations-from-mutations so you don't need to care about manually refetching.